internal_transfers/actions/src/parse.ts

```diff
@@ -33,9 +35,11 @@ export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
     // All other event emissions can be ignored for our purposes.
     const possibleSettlementEvent = tryParseSettlementEventLog(log, index);
     const possibleTransfer = tryParseERC20Transfer(log);
+    const possibleWethAction = tryParseWethAction(log);
 
-    if (possibleTransfer !== null) {
-      const transfer: TransferEvent = possibleTransfer;
+    if (possibleTransfer !== null || possibleWethAction !== null) {
```

Comment:
Here I think we should also check if the event emitter is the WETH address:
```js
possibleTransfer !== null || (eventEmitter == WETH_ADDRESS && possibleWethAction !== null)
```

`Deposit` and `Witdhrawal` look like very common event names and I expect there will be a few other unrelated contracts that could cause this branch to trigger, which would add non-existing transfers.

Reply:
is eventEmiter accessible on the log objects? I hope so.

Reply:
Addresses your comment (although in a different place). See https://github.com/cowprotocol/solver-rewards/pull/248/commits/3f34bae5e86efb2e93e3c4c9be45a689c4da4b65

Reply:
I don't think I understand your question...Did I double the code somewhere?

Reply:
Marking as resolved and added a comment in the relevant line.